File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n < 4) {
            return n;
        }

        long long previous1 = 3;
        long long previous2 = 2;
        long long sum;

        for (int i = 4; i <= n; ++i) {
            sum = previous1 + previous2;
            previous2 = previous1;
            previous1 = sum;
        }

        return sum;
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if (n < 4):
            return n
        
        previous1 = 3
        previous2 = 2
        sum = 0

        for i in range(4, n + 1):
            sum = previous1 + previous2
            previous2 = previous1
            previous1 = sum
        
        return sum
        